cmd/vochaintest/vochaintest.go

```diff
@@ -237,6 +264,78 @@ func censusImport(host string, signer *ethereum.SignKeys) {
 	log.Infof("Census created and published\nRoot: %s\nURI: %s", root, uri)
 }
 
+func setInitAccounts(treasurer, oracle, mainSigner *ethereum.SignKeys, host string) error {
+	var err error
+	log.Infof("connecting to main gateway %s", host)
+	// connecting to endpoint
+	var mainClient *client.Client
+	for tries := 10; tries > 0; tries-- {
+		mainClient, err = client.New(host)
+		if err == nil {
+			break
+		}
+		time.Sleep(1 * time.Second)
+	}
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer mainClient.Close()
+	// get the chain ID for signing the transactions
+	chainID, err := mainClient.GetChainID()
+	if err != nil {
+		log.Fatal(err)
+	}
+	// set chain id
+	treasurer.VocdoniChainID = chainID
+	oracle.VocdoniChainID = chainID
+	mainSigner.VocdoniChainID = chainID
+
+	// create accounts
+	if err := mainClient.CreateOrSetAccount(oracle, common.Address{}, "ipfs://", 0, nil); err != nil {
+		log.Fatalf("cannot create account %s: %s", oracle.Address(), err)
+	}
+	if err := mainClient.CreateOrSetAccount(mainSigner, common.Address{}, "ipfs://", 0, nil); err != nil {
+		log.Fatalf("cannot create account %s: %s", mainSigner.Address(), err)
+	}
+
+	// top up accounts
+	if err := retryMintTokensUntilSuccess(mainClient, treasurer, oracle, 10000, 5); err != nil {
+		log.Fatalf("cannot mint tokens for account %s with treasurer %s: %w", oracle.Address(), treasurer.Address(), err)
+	}
+
+	if err := retryMintTokensUntilSuccess(mainClient, treasurer, mainSigner, 10000, 5); err != nil {
+		log.Fatalf("cannot mint tokens for account %s with treasurer %s: %w", mainSigner.Address(), treasurer.Address(), err)
+	}
+
+	return nil
+}
+
+func retryMintTokensUntilSuccess(mainClient *client.Client, treasurer, account *ethereum.SignKeys, amount uint64, retries int) error {
+	// instead of waiting until a certain prerequired transaction is mined,
+	// retry the desired tx until success
+	for i := 0; i < retries; i++ {
+		treasurerAccount, err := mainClient.GetTreasurer(treasurer)
+		if err != nil {
+			return fmt.Errorf("cannot get treasurer: %w", err)
+		}
+
+		err = mainClient.MintTokens(treasurer, account.Address(), treasurerAccount.GetNonce(), amount)
```

Comment:
Also I think this can be more idiomatic with `if err := mainClient.MintTokens ...; err == nil { ... }`